### PR TITLE
WT-10307 Enable point in time truncate functionality for MongoDB

### PR DESCRIPTION
**Pull request by mm-ng**

### CODE REVIEW

**Review by korteland — COMMENTED** [review]
The code change itself looks correct, but I have some higher-level questions:
1. Has this been tested with a MongoDB patch build? If it has - are there any requirements around when we vendor this in? It might be good to split the `WT_MIN_STARTUP_VERSION` bump into another PR and let that soak for a little while before getting the truncate change in. Even if we don't care about letting them soak separately, perhaps a split makes sense in case the new truncate functionality needs to be reverted (that way we don't have to also wind back the version bump).
2. How does upgrade/downgrade testing work for this? My understanding is that the scripts will currently create databases with WT develop and try to open them with versions all the way down to MongoDB 4.4-compatible. I expect this won't work, how are we dealing with it?
3. Are there more incoming format changes? If so, will they also need to bump this number, or is this a "one and done" thing?

**Comment by kommiharibabu** [comment]
@mm-ng I doubt that the change can break older version compatibility. The older versions save the WT version as 11.1.0 or less. Upgrading those versions to the newer version after this change doesn't allow them to start. Am I missing anything?

The existing evergreen compatibility tests do not test compatibility using the patch version, as they always test using the develop branch version.

**Comment by mm-ng** [comment]
@korteland @kommiharibabu 

(1) Yes, these changes have been tested with a MongoDB patch build (which includes required tasks as well as various multiversion tests). There’s a link to it somewhere in the comments of the ticket, or I can also message it to you separate if that’s easier. The failures in the patch are known failures and unrelated to the changes.

There are requirements about when we vendor this in - it needs to go into the 8.0 version (it’s okay if we backport it to 8.0 but we shouldn’t wait until 8.1).

(2) Hmm, you might be right, they may break the MongoDB 4.4 version testing. I’ll need to test this more and perhaps find a way to run the WiredTiger compatibility/upgrade-downgrade testing locally as I forgot that the Evergreen compatibility tests use the develop branch.

(3) As of now, I don’t expect there to be anymore format changes. If there is one, I think we will need to bump the number again if we want skip-level version testing to work. If not, then I don’t think we need to bump the version. The reason why we need skip-level things to work in 8.0 is that ideally we want to allow customers to directly upgrade multiple versions (for performance reasons), and also directly downgrade multiple versions if something goes wrong with the upgrade. I don’t expect skip-level to be as important in the future.

**Comment by mm-ng** [comment]
@korteland @kommiharibabu I have run the `compatibility_test_for_releases.sh` script locally to test the older version compatibility with these changes, and looks like things are running smoothly.

File: src/reconcile/rec_write.c [hunk 1]
```diff
@@ -1942,8 +1942,7 @@ __rec_split_write_header(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_REC_CHUNK
     }
 
     /* Set the fast-truncate proxy cell information flag. */
-    if ((page->type == WT_PAGE_COL_INT || page->type == WT_PAGE_ROW_INT) &&
-      __wt_process.fast_truncate_2022)
+    if ((page->type == WT_PAGE_COL_INT || page->type == WT_PAGE_ROW_INT))
```
[the remark this answers is not in the document]
Comment (mm-ng) [reply src/reconcile/rec_write.c]: Fixed in [22dce81](https://github.com/wiredtiger/wiredtiger/pull/10433/commits/22dce81e7e2bceb515dd21b73ed8dc87f72f6791)

**Review by korteland — APPROVED** [review]
LGTM!

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Pull request merged**